sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/AnsiTypeCoercion.scala

```diff
@@ -279,31 +227,19 @@ object AnsiTypeCoercion extends TypeCoercionBase {
     override def transform: PartialFunction[Expression, Expression] = {
       // Skip nodes who's children have not been resolved yet.
       case g if !g.childrenResolved => g
-
-      case g: GetDateField if AnyTimestampTypeExpression.unapply(g.child) =>
-        g.withNewChildren(Seq(Cast(g.child, DateType)))
+      case withChildrenResolved
+          if AnsiGetDateFieldOperationsTypeCoercion.apply.isDefinedAt(withChildrenResolved) =>
+        AnsiGetDateFieldOperationsTypeCoercion.apply(withChildrenResolved)
     }
   }
 
   object DateTimeOperations extends TypeCoercionRule {
```

Comment:
while we're here, can we please rename each object that inherits (directly or indirectly) from the Catalyst Rule class to end with `Rule`, to make it simpler to differentiate which of these objects are Rules and which just contain pattern-matches?

Reply:
Done